lib/libxdp/libxdp.c

```diff
@@ -3253,8 +3290,15 @@ static int xdp_multiprog__attach(struct xdp_multiprog *old_mp,
 
 
 	err = xdp_attach_fd(prog_fd, old_fd, ifindex, mode);
-	if (err < 0)
+	if (err < 0) {
+		if (!mp) {
```

Comment:
We should only attempt the link detachment if the kernel returns EBUSY, otherwise we're just doing useless work.

Reply:
done